File: CS/MyButtonEdit/MyTextEditPainter.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;

namespace WindowsApplication1
{
    public class MyButtonEditPainter : ButtonEditPainter
    {

        public MyButtonEditPainter()
        {

        }

        public int GetScrollPosition(ControlGraphicsInfoArgs info)
        {
            object value = ((ButtonEditViewInfo)(info.ViewInfo)).EditValue;
            return Convert.ToInt32(value);
        }
        protected override void DrawContent(ControlGraphicsInfoArgs info)
        {
            DrawText(info);
        }

        protected override void DrawText(ControlGraphicsInfoArgs info)
        {
            int scrollPosition = GetScrollPosition(info);
            DevExpress.Utils.AppearanceObject appearance = info.ViewInfo.Appearance;
            Rectangle bounds = (info.ViewInfo as ButtonEditViewInfo).MaskBoxRect;
            string displayText = info.ViewInfo.DisplayText;
            bounds.Offset(0, scrollPosition);
            bounds.Height = info.Graphics.MeasureString(displayText, appearance.GetFont()).ToSize().Height;
            appearance.DrawString(info.Cache, displayText, bounds);
        }
    }
}

File: VB/MyButtonEdit/MyTextEditPainter.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo

Namespace WindowsApplication1
	Public Class MyButtonEditPainter
		Inherits ButtonEditPainter

		Public Sub New()

		End Sub

		Public Function GetScrollPosition(ByVal info As ControlGraphicsInfoArgs) As Integer
			Dim value As Object = (CType(info.ViewInfo, ButtonEditViewInfo)).EditValue
			Return Convert.ToInt32(value)
		End Function
		Protected Overrides Sub DrawContent(ByVal info As ControlGraphicsInfoArgs)
			DrawText(info)
		End Sub

		Protected Overrides Sub DrawText(ByVal info As ControlGraphicsInfoArgs)
			Dim scrollPosition As Integer = GetScrollPosition(info)
			Dim appearance As DevExpress.Utils.AppearanceObject = info.ViewInfo.Appearance
			Dim bounds As Rectangle = (TryCast(info.ViewInfo, ButtonEditViewInfo)).MaskBoxRect
			Dim displayText As String = info.ViewInfo.DisplayText
			bounds.Offset(0, scrollPosition)
			bounds.Height = info.Graphics.MeasureString(displayText, appearance.GetFont()).ToSize().Height
			appearance.DrawString(info.Cache, displayText, bounds)
		End Sub
	End Class
End Namespace
